sapphire/corsika/blocks.py

```diff
@@ -427,8 +427,12 @@ def particle_data(subblock):
     r = math.sqrt(x ** 2 + y ** 2)
     phi = math.atan2(y, x)
 
-    return (p_x, p_y, p_z, x, y, t, id, r, hadron_generation,
-            observation_level, phi)
+    if len(subblock) == 7:
+        # no thinning
+        return (p_x, p_y, p_z, x, y, t, id, r, hadron_generation, observation_level, phi)
+    else:
+        # append the thinning weight
+        return (p_x, p_y, p_z, x, y, t, id, r, hadron_generation, observation_level, phi, subblock[7])
```

Comment:
The line is too long, you should break it into multiple lines, but also need to align the new lines with the bracket of the previous line.
```
    return (p_x, p_y, p_z, x, y, t, id, r, hadron_generation,
            observation_level, phi)
```